ui/src/app/applications/components/application-sync-panel/application-sync-panel.tsx

```diff
@@ -48,7 +49,7 @@ export const ApplicationSyncPanel = ({application, selectedResource, hide}: {app
                     {isVisible && (
                         <Form
                             defaultValues={{
-                                revision: source.targetRevision || 'HEAD',
+                                revision: query.get("revision") || source.targetRevision || 'HEAD',
```

Comment:
WDYT using history from context?
```suggestion
                                revision: new URLSearchParams(ctx.history.location.search).get('revision') || source.targetRevision || 'HEAD',
```

Also for info, search query isn't cleared on modal close or navigation, so it might stay unused after.

Reply:
It looks like most places in the codebase use the `new URLSearchParams` method, so I agree with @Marvin9. I also agree about clearing this param on model close / navigation

Reply:
Thanks for the hints!

Updated the pr. Not sure if that's the way to clear the param (but in my tests it works). Couldn't find a similar example.

Reply:
Suggestion:

I found one method we can reuse. It is named `showDeploy`, we can add `revision` parameter to it and default value would be from `getAppDefaultSource(app).targetVersion` and `null` to clear it.

Reply:
great suggestion. I updated the PR to use that function